src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -268,61 +264,34 @@ protected void ThrowODataException(string errorMessage)
         protected abstract ODataBatchReaderState ReadAtChangesetEndImplementation();
 
         /// <summary>
-        /// Validate the reader state is setup properly when reader stream is at the start of changeset.
-        /// </summary>
-        protected abstract void VerifyReaderStreamChangesetStartImplementation();
-
-        /// <summary>
-        /// Reset the changeset boundary in the reader stream at the end of the changeset processing.
-        /// </summary>
-        protected abstract void ResetReaderStreamChangesetBoundary();

```

Comment:
We should keep the ReaderStreamChangeset... methods, they are tied to the changeset boundary of the reader stream hence are separate attributes from the changeset boundary of the reader itself.